src/csharp/Grpc.Core/ServerCredentials.cs

```diff
@@ -57,42 +57,106 @@ internal override ServerCredentialsSafeHandle ToNativeCredentials()
         }
     }
 
+    /// <summary>
+    /// Modes of requesting client's SSL certificate by the server.
+    /// Corresponds to <c>grpc_ssl_client_certificate_request_type</c>.
+    /// </summary>
+    public enum SslClientCertificateRequestType {
```

Comment:
Up for discussion: I don't like that the `SslClientCertificateRequestType.RequestAndRequireClientCertificateButDontVerify` literals feel extremely long, so perhaps a shorter name should be used for the type at least (ie. for `SslClientCertificateRequestType`).